includes/class-apple-news.php

```diff
@@ -211,38 +291,32 @@ public function migrate_header_settings( $wp_settings ) {
 			return $wp_settings;
 		}
 
-		// Check for presence of legacy font setting.
-		if ( ! empty( $wp_settings['header_font'] ) ) {
-			$wp_settings['header1_font'] = $wp_settings['header_font'];
-			$wp_settings['header2_font'] = $wp_settings['header_font'];
-			$wp_settings['header3_font'] = $wp_settings['header_font'];
-			$wp_settings['header4_font'] = $wp_settings['header_font'];
-			$wp_settings['header5_font'] = $wp_settings['header_font'];
-			$wp_settings['header6_font'] = $wp_settings['header_font'];
-			unset( $wp_settings['header_font'] );
-		}
-
-		// Check for presence of legacy color setting.
-		if ( ! empty( $wp_settings['header_color'] ) ) {
-			$wp_settings['header1_color'] = $wp_settings['header_color'];
-			$wp_settings['header2_color'] = $wp_settings['header_color'];
-			$wp_settings['header3_color'] = $wp_settings['header_color'];
-			$wp_settings['header4_color'] = $wp_settings['header_color'];
-			$wp_settings['header5_color'] = $wp_settings['header_color'];
-			$wp_settings['header6_color'] = $wp_settings['header_color'];
-			unset( $wp_settings['header_color'] );
-		}
-
-		// Check for presence of legacy line height setting.
-		if ( ! empty( $wp_settings['header_line_height'] ) ) {
-			$wp_settings['header1_line_height'] = $wp_settings['header_line_height'];
-			$wp_settings['header2_line_height'] = $wp_settings['header_line_height'];
-			$wp_settings['header3_line_height'] = $wp_settings['header_line_height'];
-			$wp_settings['header4_line_height'] = $wp_settings['header_line_height'];
-			$wp_settings['header5_line_height'] = $wp_settings['header_line_height'];
-			$wp_settings['header6_line_height'] = $wp_settings['header_line_height'];
-			unset( $wp_settings['header_line_height'] );
-		}
+		// Clone settings, as necessary.
+		$wp_settings = $this->_clone_settings( $wp_settings, array(
+			'header1_color' => 'header_color',
+			'header2_color' => 'header_color',
+			'header3_color' => 'header_color',
+			'header4_color' => 'header_color',
+			'header5_color' => 'header_color',
+			'header6_color' => 'header_color',
+			'header1_font' => 'header_font',
+			'header2_font' => 'header_font',
+			'header3_font' => 'header_font',
+			'header4_font' => 'header_font',
+			'header5_font' => 'header_font',
+			'header6_font' => 'header_font',
+			'header1_line_height' => 'header_line_height',
+			'header2_line_height' => 'header_line_height',
+			'header3_line_height' => 'header_line_height',
+			'header4_line_height' => 'header_line_height',
+			'header5_line_height' => 'header_line_height',
+			'header6_line_height' => 'header_line_height',
+		) );
+
+		// Remove legacy settings.
+		unset( $wp_settings['header_color'] );
+		unset( $wp_settings['header_font'] );
+		unset( $wp_settings['header_line_height'] );
```

Comment:
`unset` works whether the array key exists or not, so no need to look before I leap here.